tasksCode/AnyType.hpp

```diff
@@ -0,0 +1,412 @@
+    explicit operator unsigned int() const {
+		if (valueType != ValueType::TypeUnsignedInt)
+			throw std::bad_cast{ };
+		return static_cast<unsigned int>((*valueData).valueLLUInt);
+	};
+	
+	// Transform to unsigned short, if the internal type is unsigned short,
+	// otherwise throw bad_cast exception
+    explicit operator unsigned short int() const {
+		if (valueType != ValueType::TypeUnsignedShortInt)
+			throw std::bad_cast{ };
+		return static_cast<unsigned short>((*valueData).valueLLUInt);
+	};
+	
+	// Transform to unsigned long, if the internal type is unsigned long,
+	// otherwise throw bad_cast exception
+    explicit operator unsigned long int() const {
+		if (valueType != ValueType::TypeUnsignedLongInt)
+			throw std::bad_cast{ };
+		return static_cast<unsigned long int>((*valueData).valueLLUInt);
+	};
+	
+	// Transform to unsigned long long, if the internal type is unsigned long long,
+	// otherwise throw bad_cast exception
+    explicit operator unsigned long long int() const {
+		if (valueType != ValueType::TypeUnsignedLongLongInt)
+			throw std::bad_cast{ };
+		return static_cast<unsigned long long int>((*valueData).valueLLUInt);
+	};
+	
+	// Transform to float, if the internal type is float,
+	// otherwise throw bad_cast exception
+    explicit operator float() const {
+		if (valueType != ValueType::TypeFloat)
+			throw std::bad_cast{ };
+		return static_cast<float>((*valueData).valueLDouble);
+	};
+	
+	// Transform to double, if the internal type is double,
+	// otherwise throw bad_cast exception
+    explicit operator double() const {
+		if (valueType != ValueType::TypeDouble)
+			throw std::bad_cast{ };
+		return static_cast<double>((*valueData).valueLDouble);
+	};
+	
+	// Transform to long double, if the internal type is long double,
+	// otherwise throw bad_cast exception
+    explicit operator long double() const {
+		if (valueType != ValueType::TypeLongDouble)
+			throw std::bad_cast{ };
+		return static_cast<long double>((*valueData).valueLDouble);
+	};
+
+	// Transform to other types, not implemented in the class,
+	// generally always throw bad_cast exception
+	template <typename Type>
+    explicit operator Type() const {
+		throw std::bad_cast{ };
+	};
+
+	// Return value' type
+	std::string GetValueType() const;
+	
+	// Delete internal data
+	void DeleteData();
+	
+	// Swap two objects
+	void Swap(AnyType & anyType);
+	
+	// Swap two objects
+	friend void Swap(AnyType & anyTypeA, AnyType & anyTypeB);
+
+	// Default destructor
+	~AnyType() {
+		delete valueData;
+	}
+};
+
+// Set the default internal value' data
+void AnyType::SetValueData()
+{
+	if (valueData == nullptr)
+		valueData = new ValueData;
+	// Set all valueData bytes to 0 (the long long int has 64 bytes
+	// which is the max fundamental datatype size)
+	(*valueData).valueLLInt = 0;
+}
+
+// Set the internal value' data, based on the give value
+template <typename Type>
+void AnyType::SetValueData(const Type & value)
```

Comment:
I recommend you to unite function SetValueData() and SetValueType() to one function, SetValue() for example. It would simplify your code,because you have the same cheking in mentioned function.

Reply:
I can't because I use them separately in other functions.